File: examples/python/rabbitmq-producer-consumer/producer.py
import json
import time
import pika
import os

url = os.getenv("RABBITMQ_URL", "amqp://admin:change_me@localhost:5672/app")
params = pika.URLParameters(url)
conn = pika.BlockingConnection(params)
ch = conn.channel()

ch.exchange_declare(exchange="app.exchange", exchange_type="topic", durable=True)
ch.basic_publish(exchange="app.exchange", routing_key="app.created", body=json.dumps({"id": int(time.time())}))
print("[producer] published")
conn.close()

File: examples/nodejs/rabbitmq-producer-consumer/producer.js
import amqp from 'amqplib';

const url = process.env.RABBITMQ_URL || 'amqp://admin:change_me@localhost:5672/app';
const conn = await amqp.connect(url);
const ch = await conn.createChannel();

await ch.assertExchange('app.exchange', 'topic', { durable: true });
ch.publish('app.exchange', 'app.created', Buffer.from(JSON.stringify({ id: Date.now(), event: 'created' })));
console.log('[producer] message published');

await ch.close();
await conn.close();
